app/dao/fact_billing_dao.py

```diff
@@ -65,31 +65,40 @@ def fetch_sms_free_allowance_remainder(start_date):
 def fetch_sms_billing_for_all_services(start_date, end_date):
 
     # ASSUMPTION: AnnualBilling has been populated for year.
-    free_allowance_remainder = fetch_sms_free_allowance_remainder(start_date).subquery()
+    allowance_left_at_start_date_query = fetch_sms_free_allowance_remainder_until_date(start_date).subquery()
```

Comment:
How does this fix the bug? We are still passing in the `start_date` rather than `the end_date`

Reply:
yeah, sorry, should have split this into commits more, I was renaming things to help myself make sense of what is happening as I was problem solving, and that was a bit of a messy process, hence single commit 😅 

As @benthorner said, `sms_allowance_left` is what we return as `sms_remainder` for the report, while `allowance_left_at_start_date` is still needed because we use it to calculate `chargeable_sms`.